### DISCUSSION TITLE
Can URLs support multiple? The same link supports multiple settings

### DISCUSSION THREAD

**Discussion by sunkaifei** [actix/actix-web#3435] [Q&A] [answered]
Can URLs support multiple? The same link supports multiple settings, such as /abc and /abc/ 

java spring
```java
@GetMapping(value = {"/a-{area}/" , "/a-{area}-{p}/"})
```

Spring's routing can be set in array form to point to a single controller, so there is no need to write multiple controllers, such as an article with comments and page flipping, or a path with/or without

**Comment by robjtede** [comment]
You might be looking for the [`#[routes]`](https://docs.rs/actix-web/latest/actix_web/attr.routes.html) macro.

**Reply by sunkaifei** [reply]
I tried to take a look, but it seems like I haven't directly transformed it into an array for visual purposes. For example:

`
#[get({"/test"，"/abc"})]
`

**Reply by robjtede** [reply, marked as answer]
There's no macro that has that syntax. You best solution using macros is:

```rust
#[routes]
#[get("/test")]
#[get("/abc")]
async fn handler() -> { ... }
```

**Reply by sunkaifei** [reply]
Thank you very much for your help

**Comment by robjtede** [comment]
Using manual routing is also an option, the [`Resource::new` constructor](https://docs.rs/actix-web/4/actix_web/struct.Resource.html#method.new) accepts multiple paths.

```rust
Resource::new(["/test", "/abc"])
```
